src/core/common/message.hpp

```diff
@@ -217,6 +217,7 @@ class Buffer : public otMessageBuffer, public LinkedListEntry<Buffer>
         bool    mDoNotEvict : 1;       // Whether this message may be evicted.
         bool    mMulticastLoop : 1;    // Whether this multicast message may be looped back.
         bool    mResolvingAddress : 1; // Whether the message is pending an address query resolution.
+        uint8_t mOrigin : 2;           // The origin of the message.
```

Comment:
In `MessagePool::Allocate()` when we allocate a new message the whole message is `memset` to zero and then certain fields are set (calling `SetType`, `SetLinkSecurityEnabled`, `SetPriority()` ect). Everything else is cleared.

This means all message when they are first allocated have `mOrigin` as zero by default which maps to `kOriginThreadNetif` This as default make sense to me since we mostly allocate for use within OT stack itself (we are either receiving it or sending something). 

When we pass the allocated message to user from one of OT public APIs we may want to change the origin. I think these are the APIs where we allocate new message and give it to host to use:
- `otCoapNewMessage()`
- `otIp6NewMessage`
- `otIp6NewMessageFromBuffer()`
- `otIp4NewMessage()`
- `otUdpNewMessage()`